tests/testsuite/doc.rs

```diff
@@ -2697,3 +2697,40 @@ fn doc_lib_false_dep() {
     assert!(p.build_dir().join("doc/foo").exists());
     assert!(!p.build_dir().join("doc/bar").exists());
 }
+
+#[cargo_test]
+fn link_to_private_item() {
+    let main = r#"
+    //! [bar]
+    #[allow(dead_code)]
+    fn bar() {}
+    "#;
+    let p = project()
+        .file("src/lib.rs", main)
+        .file("src/main.rs", main)
+        .build();
+    p.cargo("doc --lib")
+        .with_stderr_contains("[..] documentation for `foo` links to private item `bar`")
+        .run();
```

Comment:
Sorry for the delay.

I would avoid the filename collision and avoid the issue with output checks by essentially doing separate projects. This can be done by just renaming the file and running `cargo doc` again.  Something like this:



```suggestion
    let p = project().file("src/lib.rs", main).build();
    p.cargo("doc")
        .with_stderr_contains("[..] documentation for `foo` links to private item `bar`")
        .run();
    // Check that binaries don't emit a private_intra_doc_links warning.
    fs::rename(p.root().join("src/lib.rs"), p.root().join("src/main.rs")).unwrap();
    p.cargo("doc")
        .with_stderr(
            "[DOCUMENTING] foo [..]\n\
             [FINISHED] [..]",
        )
        .run();
```

Reply:
My turn to apologize for the delay :grin: I agree your test is much better. I applied your patch and the test is passing reliably now.